src/ClaimAction.sol

```diff
@@ -223,19 +174,42 @@ contract ClaimActionUpgradeable is
         forwardingAddresses[msg.sender] = forwardingAddress;
     }
 
-    /// @notice Enable/disable balancer sandwich prevention
-    /// @param preventSandwiching true: sandwich prevention enabled in the balancer swap step
-    function changeBalancerSandwichPrevention(
-        bool preventSandwiching
-    ) public onlyOwner {
-        balancerSandwichPrevention = preventSandwiching;
+    function chainlinkGnoEurPrice() public view returns (uint256) {
+        uint256 gnoUsd = uint256(getChainlinkGnoUsdDataFeedLatestAnswer());
+        uint256 eurUsd = uint256(getChainlinkEurUsdDataFeedLatestAnswer());
+        require(eurUsd > 0, "EUR/USD feed is 0");
+        return (gnoUsd * 1e8) / eurUsd;
+    }
+
+    function getChainlinkGnoUsdDataFeedLatestAnswer()
+        public
+        view
+        returns (int)
+    {
+        // prettier-ignore
+        (
+            /* uint80 roundID */,
+            int answer,
+            /*uint startedAt*/,
+            /*uint timeStamp*/,
+            /*uint80 answeredInRound*/
+        ) = gnoUsdFeed.latestRoundData();
```

Comment:
Can you check that `answer > 0` and `block.timestamp - answerTimestamp < 1 days` to ensure the oracle data is fresh and not empty (for both functions)?